### PR TITLE
[COR-891] updating quimby

### PR DESCRIPTION
**Pull request by apynes**
- fix writer not respecting bytes written
- fixing some ignored errors
- allow for parsing a multipart form into an object
- updating dependencies

### CODE REVIEW

**Review by anthonyheidenreich — APPROVED** [review]

File: http/context_test.go [hunk 1]
```diff
@@ -644,12 +648,77 @@ func TestReadObject_withForm_EmptyQueryParam(t *testing.T) {
...
-	context.Write("foo")
+	assert.NoError(context.Write([]byte("foo")))
 	assert.Equal("foo", stringutil.NullTerminatedString(writerBody))
 }
+
+func Test_context_valuesToObject(t *testing.T) {
+	assert := assert.New(t)
+
+	v := url.Values{}
+	v.Add("name", "RoundyMcTrashcan")
```
Comment (nmarie22) [inline http/context_test.go:665]: https://www.youtube.com/watch?v=n0XaSvhTYd4

File: http/context_test.go
Comment (nmarie22) [inline http/context_test.go:1]: The test `TestReadObject_withJSON_Empty` is failing for me

File: http/multipartform/testdata/formencodeddata.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+I am a dimunitive dispenser of flavor infused water
+--xYzZY
+Content-Disposition: form-data; name="my_array[]"
+
+short and stout
+--xYzZY--
+`
+
+	FormDataWithStructs = `--xYzZY
```
Comment (nmarie22) [inline http/multipartform/testdata/formencodeddata.go:27]: Missing doc string

File: http/multipartform/testdata/formencodeddata.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+{ "Foo": "qux", "Bar": 1, "Baz": ["elm1", "elm2"] }
+--xYzZY
+Content-Disposition: form-data; name="test_struct"
+
+{ "Foo": "quux", "Bar": 2, "Baz": ["1mle", "2mle"]}
+--xYzZY--
+`
+
+	FormDataErryThang = `--xYzZY
```
Comment (nmarie22) [inline http/multipartform/testdata/formencodeddata.go:39]: Missing doc string

**Review by nmarie22 — APPROVED** [review]

File: http/context_test.go (hunk 1 above)
Comment (apynes) [reply http/context_test.go:665]: :wastebasket:

**Pull request merged**